knowhere/index/vector_index/IndexDiskANNConfig.cpp

```diff
@@ -0,0 +1,215 @@
+// Copyright (C) 2019-2020 Zilliz. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance
+// with the License. You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software distributed under the License
+// is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+// or implied. See the License for the specific language governing permissions and limitations under the License.
+
+#include "knowhere/index/vector_index/IndexDiskANNConfig.h"
+
+#include <limits>
+#include <optional>
+
+#include "index/vector_index/ConfAdapter.h"
+
+namespace knowhere {
+namespace {
+static constexpr const char* kDataPath = "data_path";
+static constexpr const char* kMaxDegree = "max_degree";
+static constexpr const char* kSearchListSize = "search_list_size";
+static constexpr const char* kSearchDramBudgetGb = "search_dram_budget_gb";
+static constexpr const char* kBuildDramBudgetGb = "build_dram_budget_gb";
+static constexpr const char* kNumThreads = "num_threads";
+static constexpr const char* kPqDiskBytes = "pq_disk_bytes";
+
+static constexpr const char* kNumNodesToCache = "num_nodes_to_cache";
+static constexpr const char* kWarmUp = "warm_up";
+static constexpr const char* kUseBfsCache = "use_bfs_cache";
+
+static constexpr const char* kK = "k";
+static constexpr const char* kBeamwidth = "beamwidth";
+static constexpr const char* kCandidatesLimit = "candidates_limit";
```

Comment:
Does it limit anything? I saw a reference in `full_retset.reserve(candidates_limit)`, but it seems just a preallocate.

Reply:
Good catch! I missed that. will delete it.